### PR TITLE
Say cannot be accessed as a member, cf dotc

### LINKED ISSUE #11680: Clarify message for inaccessible method
When an `object Foo` attempts to access a `private` method of an `object Bar`, the compiler says that the method cannot be accessed "in object Bar", rather than "in object Foo". For example,

contents of `Test.scala`:

```scala
object Bar {
  private def foo: String = "foo"
}

object Foo {
  def test: String = Bar.foo
}
```

output of `scalac Test.scala` (for both 2.13.0 and 2.12.9):

```scala
Test.scala:6: error: method foo in object Bar cannot be accessed in object Bar
  def test: String = Bar.foo
                         ^
one error found
```

### PR DESCRIPTION
**Pull request by som-snytt**
Also fix `partest --grep "string in check file"`.

Fixes scala/bug#11680

### CODE REVIEW

**Review by NthPortal — APPROVED** [review]
I'm not entirely certain what's going on in the `PartestUtil` changes, but otherwise looks good to me.

I wouldn't say it closes the bug though, because I think it would be good to have parity with Dotty and add a "from" clause to the end

**Comment by som-snytt** [comment]
This is more like Dotty parody than Dotty parity.

File: project/PartestUtil.scala [hunk 1]
```diff
@@ -58,10 +58,19 @@ object PartestUtil {
     val Grep = {
       def expandGrep(x: String): Seq[String] = {
         val matchingFileContent = try {
-          val Pattern = ("(?i)" + x).r
+          val Pattern = raw"(?i)$x".r
```
Comment (retronym) [inline project/PartestUtil.scala]: `${Pattern.quote(x)}` would also defend against regex special characters in the input.

Comment (som-snytt) [reply project/PartestUtil.scala]: Tests of a broken status quo should have the form, `t555?.scala`.

Comment (som-snytt) [reply project/PartestUtil.scala]: I wrote a special test file, but guess what, it turns out there was already a test with the string `hello?`.

**Comment by som-snytt** [comment]
Because I cannot deny either of you, the follow-up addresses your concerns and requests.

**Pull request merged**